web/src/user_events.ts

```diff
@@ -131,6 +132,11 @@ export const update_person = function update(event: UserUpdate): void {
             settings_realm_user_settings_defaults.maybe_disable_widgets();
             settings_account.update_account_settings_display();
             settings.update_lock_icon_in_sidebar();
+            settings_account.update_user_self_role_select_state();
```

Comment:
We should also add the code for actually updating the role dropdown when role of a user is changed.